### DISCUSSION TITLE
Possible add a second link attribute in object list?

### DISCUSSION THREAD

**Discussion by zsofiaj** [thegetty/quire#794] [Get Help]
In our collection catalogue, we have a few entries that catalogue two objects. The tombstone data for these two objects are combined, and the two images are combined in a single image. Has anyone tried adding a second link attribute in the objects.yaml file? We would like two links to show on these pages to take users to the individual objects in our online database.

Here is an example (with the combined tombstone, but only one link showing, which takes you to the first of the two objects):

https://italian-paintings-march2023-updates.netlify.app/catalogue/volume-1/the-florentine-school/24b/

I tried entering the second object in the objects.yaml file, with only the link attribute filled in, the rest blank, and adding the object id to the entry md file, and that worked in adding the second link in my preview, but when I tried to push the edits to the preview site, the deploy failed. If it is not possible to add the second link button, is there a way to somehow manually add two buttons that would mimic the look of the link button?

**Comment by Erin-Cecele** [comment]
Hi @zsofiaj, Thank you for this inquiry. Quire was built to accommodate one object image, one tombstone, and one "view in collection" link per object entry. However, the predicament of how to handle object entries with multiple pieces also came up recently with the Michael C. Carlos Museum.

Unfortunately, as of now, there is no quick and easy solution. To achieve what you propose above, you would need to do some in-depth customization of the javascript in the templates. This is what freelance developer, @chrisdaaz, had to do to enable multiple tombstones in the recent Michael C. Carlos Museum collection catalogue, as you can see here: https://carlos.emory.edu/publications/LifeAndTheAfterlife/catalog/daily-life-in-ancient-egypt/3/. Would you like to look into developer support for your catalogue?

Looking forward, accommodating multiple images, tombstones, and collection links per object entry is a capability we would love to see added to Quire at some point. While it is not on our roadmap at Getty, this is the sort of thing we would heartily encourage our community members to develop and contribute back to Quire.

**Reply by zsofiaj** [reply]
Thanks so much, @Erin-Cecele  I was actually able to make each page show multiple tombstones, but then the curator decided he wanted to combine two tombstones into one, which is where I ran into trouble. So we just have one tombstone for multiple objects, which made it impossible to add the second link. Is there a way to manually create an additional button that looks like the "view in collection" button? If not, I'll discuss with Tiffany to see if she'd like to involve a developer. I'm hoping we can figure out a workaround, though.

**Reply by chrisdaaz** [reply]
>that worked in adding the second link in my preview, but when I tried to push the edits to the preview site, the deploy failed

What was the failure message in the deploy logs? Do you get the same error running `quire build` locally? That might give you a starting point for finding a workaround. If you can build it locally just fine, use Netlify Drop to host the preview files and deploy the site manually to production when you're ready. Maybe it's just a Netlify issue?

**Worst case scenario:** you run `quire build` locally and edit the raw HTML for the page to manually hardcode a second link. Truly a bad a idea that will get messy if you ever need to run `quire build` again because each subsequent build will erase that second link. You'd have to keep good documentation on how and when to make these edits for each object because it's adding some manual tasks for each site update. I probably shouldn't even mention it. :)

```html
<div class="container">
    <table class="table is-fullwidth">
    ...
     </table>                  
     <a class="button" href="[Insert Link 1]" target="_blank">
                    View in Collection
         <svg data-outputs-exclude="epub,pdf">
            <switch><use xlink:href="#link-icon"></use></switch>
          </svg>
       </a>
        <a class="button" href="[Insert Link 2]" target="_blank">
                    View in Collection
           <svg data-outputs-exclude="epub,pdf">
              <switch><use xlink:href="#link-icon"></use></switch>
            </svg>
        </a>
</div>
```

**Reply by zsofiaj** [reply]
Thanks so much, @chrisdaaz . The error messages in the deploy log looked like this:

12:51:57 PM: [quire] WARN	 shortcodes:figure              The figure id null was found in the template ./content/catalogue/volume-1/the-florentine-school/24b.md, but is not defined in figures.yaml

AND maybe also this?

12:52:21 PM: [vite:build-html] EISDIR: illegal operation on a directory, read
12:52:21 PM: file: /opt/build/repo/.11ty-vite/catalogue/volume-1/the-florentine-school/24b/index.html
12:52:21 PM: [11ty] Encountered a Vite build error, restoring original Eleventy output to _site [Error: EISDIR: illegal operation on a directory, read] {
12:52:21 PM:   errno: -21,
12:52:21 PM:   code: 'PLUGIN_ERROR',
12:52:21 PM:   syscall: 'read',
12:52:21 PM:   pluginCode: 'EISDIR',
12:52:21 PM:   plugin: 'vite:build-html',
12:52:21 PM:   hook: 'transform',
12:52:21 PM:   id: '/opt/build/repo/.11ty-vite/catalogue/volume-1/the-florentine-school/24b/index.html',

But I'm not totally sure if these are both relevant?

And thank you so much for the html idea, but I don't think I'll be able to use this workaround because this is a project that will need to be uploaded countless times. Perhaps it was a foolish idea, but we are adding entries in batches, and there have been many rounds of edits. I wish there were a way to add a button right in the markdown file for the entry.
